src/price/dto/update-price.dto.ts

```diff
@@ -0,0 +1,19 @@
+import { IsNotEmpty, IsArray, IsOptional, IsObject } from "class-validator";
+export class UpdatePriceDto {
+  @IsObject()
+  @IsOptional()
+  @IsNotEmpty()
+  readonly tiles: {
+    tile_id: string;
+    deliveryPrice: number;
+    tilePrice: { color: string; price: number }[];
+  };
+
+  @IsArray()
+  @IsOptional()
+  @IsNotEmpty()
+  readonly court_spec: {
+    court: string;
+    installationPrice: number;
+  }[];
+}

```

Comment:
I think maybe you can make this inherit from create-dto if these two have the similar structure and content.